### PR TITLE
pytorch-lightning 2.3.0

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Destination channel:** main

### Links

- [PKG-2731](https://anaconda.atlassian.net/browse/PKG-2731) 
- [Upstream repository](https://github.com/Lightning-AI/pytorch-lightning/tree/2.3.0)
- Requirements:
  - https://github.com/Lightning-AI/pytorch-lightning/blob/2.3.0/pyproject.toml
  - https://github.com/Lightning-AI/pytorch-lightning/blob/2.3.0/requirements.txt
  - https://github.com/Lightning-AI/pytorch-lightning/blob/2.3.0/requirements/pytorch/base.txt
  - 

### Explanation of changes:

- Update runtime pinnings

### Notes:

- Updating because of the `pytorch 2.3.0` release

[PKG-2731]: https://anaconda.atlassian.net/browse/PKG-2731?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml
Comment (danpetry) [inline recipe/meta.yaml]: you don't need upper bounds, see notes in requirements/pytorch/base.txt

**Review by danpetry — COMMENTED** [review]
could you please add the test suite?

File: recipe/meta.yaml
Comment (skupr-anaconda) [reply recipe/meta.yaml:25]: The Source Distribution on PyPI is missing the tests folder, see https://pypi.org/project/pytorch-lightning/#files
I tried to test it from the upstream github repo but the tests are quite heavy.

Another issue is that the Source Distribution on PyPI installs lightning_fabric and pytorch_lightning but if you install from github upstream tarballs then lightning will be installed but pytorch_lightning won't be available because of another structure. That will break the user experience.
I would follow the conda-forge approach using the PyPI source

Comment (skupr-anaconda) [reply recipe/meta.yaml]: The Source Distribution on PyPI is missing the tests folder, see https://pypi.org/project/pytorch-lightning/#files
I tried to test it from the upstream github repo but the tests are quite heavy.

Another issue is that the Source Distribution on PyPI installs `lightning_fabric` and `pytorch_lightning` but if you install from github upstream tarballs then `lightning` will be installed but `pytorch_lightning` won't be available because of another structure. That will break the user experience.
I would follow the `conda-forge` approach using the PyPI source

**Review by JamesRobertsonGames — APPROVED** [review]

File: recipe/meta.yaml
Comment (skupr-anaconda) [reply recipe/meta.yaml]: Removed upper bounds

Comment (danpetry) [reply recipe/meta.yaml]: ah, ok. well, it's a pure python package isn't it. fine with me then

File: recipe/meta.yaml
Comment (danpetry) [inline recipe/meta.yaml]:
```suggestion
    - fsspec >=2022.5.0
```

File: recipe/meta.yaml
Comment (danpetry) [inline recipe/meta.yaml]:
```suggestion
    - packaging >=20.0
```

File: recipe/meta.yaml
Comment (danpetry) [inline recipe/meta.yaml:34]: Maybe add a note in that we're not supporting lightning.app at the moment. You could also update [lightning-cloud](https://github.com/AnacondaRecipes/lightning-cloud-feedstock/blob/main/recipe/meta.yaml) and include the required deps, if you want, but I'll leave that up to you

**Review by danpetry — APPROVED** [review]
approving contingent on below comments being addressed

File: recipe/meta.yaml
Comment (skupr-anaconda) [reply recipe/meta.yaml:34]: I left a comment about the source of the package's dependencies

**Pull request merged**
